File: base/src/main/kotlin/ch/patchcode/pr3/logbook/objects/Facility.kt
package ch.patchcode.pr3.logbook.objects

data class Facility(
		val id: Long,
		val game: Game,
		val name: String,
		val constructionCost: Int = 0,
		val constructionDays: Int = 0,
		val maintenancePerDay: Int = 0
)

data class FacilityConsumption(
		val facility: Facility,
		val good: Good,
		val amount: Int
)

data class FacilityProduction(
		val facility: Facility,
		val good: Good,
		val amount: Int
)

File: angular-gui/src/app/games/game/facilities/facility.ts
import { Turnover } from './facility/turnover';

export class Facility {

  constructor(
    public id: number,
    public name: string
  ) {}

  constructionCost: number;
  constructionDays: number;
  maintenancePerDay: number;
  workers: number;
  material: Turnover[];
  consumption: Turnover[];
  production: Turnover;
}
